rts/Game/Camera/SpringController.cpp

```diff
@@ -54,13 +63,44 @@ void CSpringController::ConfigUpdate()
 	fastScaleMousewheel = configHandler->GetFloat("CamSpringFastScaleMousewheelMove");
 	doRotate = configHandler->GetBool("CamSpringEdgeRotate");
 	lockCardinalDirections = configHandler->GetBool("CamSpringLockCardinalDirections");
+	trackMapHeight = configHandler->GetInt("CamSpringTrackMapHeightMode");
+
+	if (trackMapHeight == HeightTracking::Smooth && !modInfo.enableSmoothMesh) {
+		LOG_L(L_ERROR, "Smooth mesh disabled");
+		trackMapHeight = HeightTracking::Terrain;
+	}
 }
 
 void CSpringController::ConfigNotify(const std::string & key, const std::string & value)
 {
 	ConfigUpdate();
 }
 
+void CSpringController::SmoothCamHeight(const float3& prevPos) {
+	if (!pos.IsInBounds()) {
+		return;
+	}
+
+	const float camHeightDiff = (trackMapHeight == HeightTracking::Smooth) ?
+		smoothGround.GetHeight(pos.x, pos.z) - smoothGround.GetHeight(prevPos.x, prevPos.z) :
+		0.0f;
+
+	if (trackMapHeight == HeightTracking::Smooth || trackMapHeight == HeightTracking::Disabled) {
```

Comment:
Looks suspicious that not only is `HeightTracking::Disabled` handled in some way but also it's in the function supposedly dedicated for `HeightTracking::Smooth`.